client/components/search/index.jsx

```diff
@@ -298,7 +307,7 @@ const Search = React.createClass( {
 				<input
 					type="search"
 					id={ 'search-component-' + this.state.instanceId }
-					className={ 'search__input' + ( this.props.dir ? ' ' + this.props.dir : '' ) }
+					className={ 'search__input' + isCloseButtonVisible + ( this.props.dir ? ' ' + this.props.dir : '' ) }
```

Comment:
I strongly recommend using [`classnames`](https://github.com/JedWatson/classnames#usage) instead of relying on patching together strings :) The resulting code is usually much more readable and extendable.